dev/run-tests.py

```diff
@@ -122,9 +123,15 @@ def determine_modules_to_test(changed_modules):
     # If we need to run all of the tests, then we should short-circuit and return 'root'
     if modules.root in modules_to_test:
         return [modules.root]
-    return toposort_flatten(
+    changed_modules = toposort_flatten(
         {m: set(m.dependencies).intersection(modules_to_test) for m in modules_to_test}, sort=True)
 
+    # TODO: Skip hive-thriftserver module for hadoop-3.2. remove this once hadoop-3.2 support it
```

Comment:
Could we revert modules.py to https://github.com/apache/spark/blob/3729efb4d0420700a396c79a83a1d5db25ac3bcb/dev/sparktestsupport/modules.py and just keep this change is enough?

Reply:
Just keeping this change isn't enough because build and some other checks use profiles from `root`'s `build_profile_flags`.

Reply:
OK. Thank you @HyukjinKwon

Reply:
To be clear, this fix itself is a temp fix too in order to make the affected scope isolated and minimised. To fix it properly, we should maybe be able to fix it within `Module` at `modules.py`.